src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -300,11 +302,19 @@ def handle_message(self, message: Message) -> None:
                                 "Unable to fetch entity from query in message"
                             )
                         entity_key = entity_match.group(2)
-                    _delete_from_snuba(
-                        self.topic_to_dataset[message.topic()],
-                        contents["subscription_id"],
-                        EntityKey(entity_key),
-                    )
+                    topic = message.topic()
+                    if topic in self.topic_to_dataset:
+                        _delete_from_snuba(
+                            self.topic_to_dataset[topic],
+                            contents["subscription_id"],
+                            EntityKey(entity_key),
+                        )
+                    else:
+                        logger.error(
+                            "Topic not registered with QuerySubscriptionConsumer, can't remove "
+                            "non-existent subscription from Snuba",
+                            extra={"topic": topic, "subscription_id": contents["subscription_id"]},
+                        )
```

Comment:
Just makes sure that we don't crash the whole consumer if we haven't added the topic to `self.topic_to_dataset`. It's not a huge deal if we can't delete the subscription from snuba immediately, and we're better off continuing processing results.